src/System.Buffers/src/System/Buffers/NativeBufferPool.cs

```diff
@@ -21,6 +40,13 @@ public NativeBufferPool(int maxElementsInBuffer)
             _buckets = new NativeBufferBucket<T>[maxBuckets + 1];
         }
 
+        public void Dispose()
+        {
+            foreach (NativeBufferBucket<T> bucket in _buckets)
```

Comment:
In my example, I set `_buckets` to null. This guarantees that the native allocations are disposed *once* and it is lock free.